### DISCUSSION TITLE
Feature request: mouse gestures

### DISCUSSION THREAD

**Discussion by giaaaacomo** [zen-browser/desktop#2034] [Ideas]
i'm really used to gestures like right click+swipe left/right to go back/forward and it's so fast and useful that it would be a really nice addition

**Comment by giaaaacomo** [comment]
it would also be nice something like right click+middleclick drag up and down to rapidly move through tabs

**Comment by Tobitosan** [comment]
I wanted to completely migrate from Vivald to Zen, but the gesture navigation you have in Edge and Vivald makes a big difference in productivity. 
This is the decisive functionality for my migration. There are extensions in the store, like Gesturefy, that allow you to use these gestures, but they don't work on all tabs and this ends up disrupting my workflow.

**Comment by leic4u** [comment]
Gesturefy does not work on restricted pages. Please consider adding built-in mouse gestures to Zen Browser to support mouse gestures on all pages.

**Reply by Felix14-v2** [reply]
Another problem with Gesturefy is that it is unable to work on tabs before the website is loaded. Which leads to a "softlock" when trying to open an unreachable website or `about: ...`. pages. This is a fundamental browser limitation and can not be solved on the extension side.
